alarm_trap_sender.cpp

```diff
@@ -50,35 +50,61 @@ AlarmFilter AlarmFilter::_instance;
 
 AlarmTrapSender AlarmTrapSender::_instance;
 
-bool ActiveAlarmList::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
+bool ObservedAlarms::update(AlarmTableDef& alarm_table_def, const std::string& issuer)
 {
-  bool updated = false;
+  bool send_trap = false;
 
   std::map<unsigned int, AlarmListEntry>::iterator it = _index_to_entry.find(alarm_table_def.alarm_index());
-
+  
+  // Enter this if statement if the alarm we are currently raising is being
+  // raised with a non-CLEARED severity. 
   if (alarm_table_def.is_not_clear())
```

Comment:
Yes I do, the difference with the refactored ObservedAlarms is that it keeps track of the last state in which an alarm was raised, whether that be CLEARED or otherwise (as opposed to ActiveAlarmList which only kept track of alarms raised with a non-CLEARED severity).
That means there is no need to treat CLEARED alarms and non-CLEARED alarms differently, we just add them to the mapping if either they are not already in there, or they are but in a different state.